relayer-cli/src/commands/query/channels.rs

```diff
@@ -1,60 +1,184 @@
+        let port_id = identified_channel.port_id;
+        let channel_id = identified_channel.channel_id;
+        let chain_id = chain_id.clone();
+        let channel_end = identified_channel.channel_end;
+
+        if channel_end.state_matches(&State::Uninitialized) {
+            return Err(format!(
+                "{}/{} on chain {} @ {:?} is uninitialized",
+                port_id, channel_id, chain_id, chain_height
+            )
+            .into());
+        }
+
+        let connection_id = channel_end
+            .connection_hops
+            .first()
+            .ok_or_else(|| {
+                format!(
+                    "missing connection_hops for {}/{} on chain {} @ {:?}",
+                    port_id, channel_id, chain_id, chain_height
+                )
+            })?
+            .clone();
+
+        let connection_end = chain.query_connection(&connection_id, chain_height)?;
+        let client_id = connection_end.client_id().clone();
+        let client_state = chain.query_client_state(&client_id, chain_height)?;
+        let counterparty_chain_id = client_state.chain_id.clone();
+
+        match &cmd.destination_chain {
+            Some(dst_chain_id) if dst_chain_id != &counterparty_chain_id => {
+                continue;
+            }
+            _ => { /* proceed */ }
+        }
+
+        if !cmd.verbose {
+            output.push(PortChannelId {
+                channel_id,
+                port_id,
+            });
+        } else {
+            let channel_counterparty = channel_end.counterparty().clone();
+            let connection_counterparty = connection_end.counterparty().clone();
+            let counterparty_client_id = connection_counterparty.client_id().clone();
+
+            let counterparty_connection_id = connection_counterparty.connection_id.ok_or_else(|| {
+                format!(
+                    "connection end for {} on chain {} @ {:?} does not have counterparty connection id: {:?}",
+                    connection_id,
+                    chain_id,
+                    chain_height,
+                    connection_end
+                )
+            })?;
+
+            let counterparty_port_id = channel_counterparty.port_id().clone();
+
+            let counterparty_channel_id = channel_counterparty.channel_id.ok_or_else(|| {
+                format!(
+                    "channel end for {}/{} on chain {} @ {:?} does not have counterparty channel id: {:?}",
+                    port_id, channel_id, chain_id, chain_height, channel_end
+                )
+            })?;
+
+            let chain_config_b = config.find_chain(&counterparty_chain_id).ok_or_else(|| {
+                format!(
+                    "counterparty chain '{}' not found in configuration file",
+                    counterparty_chain_id
+                )
+            })?;
+
+            let counterparty_chain = CosmosSdkChain::bootstrap(chain_config_b.clone(), rt.clone())?;
```

Comment:
Since this is inside the loop, would it cause issues if the same chain is bootstrapped multiple times? Can we use something like `Registry` to help manage the chain runtimes?

Reply:
https://github.com/informalsystems/ibc-rs/pull/1243/commits/78a617c6313a4b733b72b7b32fd8677ac79f6d40